src/WebJobs.Script/Diagnostics/ILoggerFactoryBuilder.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Provides ways to plug into the ScriptHost ILoggerFactory initialization.
+    /// </summary>
+    public interface ILoggerFactoryBuilder
```

Comment:
FYI -- I have some more requirements coming in from elsewhere on ILogger customization so the interface may change fairly soon. Any objection to this going in as-is so that I can unblock my E2E tests that have been sitting for a while?